### PR TITLE
Add provider/pool create from memspace functions

### PR DESCRIPTION
**Pull request by kswiecicki**
TODO:
- [x] add umfMemspaceCreateFromNumaId
- [x] add tests

### CODE REVIEW

File: src/memory_pool_default.c [hunk 1]
```diff
@@ -68,3 +74,7 @@ umfPoolGetMemoryProvider(umf_memory_pool_handle_t hPool,
 
     return UMF_RESULT_SUCCESS;
 }
+
+void umfPoolSetOwnProvider(umf_memory_pool_handle_t hPool) {
```
Comment (igchor) [inline src/memory_pool_default.c]: Is this used anywhere?

File: src/memory_targets/memory_target_numa.c [hunk 2]
```diff
@@ -32,27 +34,81 @@ static enum umf_result_t numa_initialize(void *params, void **mem_target) {
...
+        .traces = 0, // TODO: parse env variable for log level?
+};
 
 static enum umf_result_t numa_memory_provider_create_from_memspace(
-    umf_memspace_handle_t memspace, void **mem_targets,
+    umf_memspace_handle_t memspace, void **mem_targets, size_t num_targets,
     umf_memspace_policy_handle_t policy,
     umf_memory_provider_handle_t *provider) {
-    // TODO: cast mem_targets to (struct numa_memory_target_t**), parse ids and initialize os provider
+    // TODO: validate that provided targets belong to the memspace?
```
Comment (igchor) [inline src/memory_targets/memory_target_numa.c]: This is actually done at a higher level right now (in memspace.c)

File: src/memory_targets/memory_target_numa.c [hunk 3]
```diff
@@ -32,27 +34,81 @@ static enum umf_result_t numa_initialize(void *params, void **mem_target) {
...
     (void)memspace;
-    (void)mem_targets;
+    // TODO: apply policy
     (void)policy;
-    (void)provider;
-    return UMF_RESULT_ERROR_NOT_SUPPORTED;
+
+    // TODO: retrieve targets from mespace instead of passing as a param?
```
Comment (igchor) [inline src/memory_targets/memory_target_numa.c]: To do this, I think we would need to have something like: https://github.com/oneapi-src/unified-runtime/pull/446 Without this, memory_target would need to know how `umf_memory_target_t` is implemented and we don't want to expose that.

File: src/memory_targets/memory_target_numa.c [hunk 4]
```diff
@@ -32,27 +34,81 @@ static enum umf_result_t numa_initialize(void *params, void **mem_target) {
...
+
+    // Create node mask from IDs
+    unsigned long nodemask = 0;
+    for (size_t i = 0; i < num_targets; i++) {
+        nodemask |= (1UL << numa_targets[i]->id);
+    }
+
+    umf_os_memory_provider_params_t params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_DEFAULT;
+    params.nodemask = &nodemask;
```
Comment (igchor) [inline src/memory_targets/memory_target_numa.c:76]: should you also set maxnode?

Comment (kswiecicki) [reply src/memory_targets/memory_target_numa.c:76]: Done.

File: src/memory_pool_default.c (hunk 1 above)
Comment (kswiecicki) [reply src/memory_pool_default.c]: No, I've removed it.

**Comment by igchor** [comment]
There are some failure on the CI - could you fix those?

File: include/umf/memspace.h [hunk 5]
```diff
@@ -50,6 +50,12 @@ umfMemoryProviderCreateFromMemspace(umf_memspace_handle_t hMemspace,
                                     umf_memspace_policy_handle_t hPolicy,
                                     umf_memory_provider_handle_t *hProvider);
 
+///
+/// \brief Creates new memory provider from memspace and policy.
```
Comment (igchor) [inline include/umf/memspace.h:54]: Wrong `brief`

File: src/memory_targets/memory_target_numa.c (hunk 4 above)
Comment (igchor) [reply src/memory_targets/memory_target_numa.c:76]: I think it's still not done. I don't see 'params.maxnode = ...' anywhere.

**Review by igchor — COMMENTED** [review]
LGTM, just 2 minor issues

**Pull request merged**